sdk/core/core-client/test/serviceClient.spec.ts

```diff
@@ -294,6 +294,52 @@ describe("ServiceClient", function () {
     assert.strictEqual(rawResponse?.headers.get("X-Extra-Info"), "foo");
   });
 
+  it("should call onResponse with the full response when encountering an unknown status", async function () {
+    let request: OperationRequest;
+
+    const pipeline = createEmptyPipeline();
+    pipeline.addPolicy(deserializationPolicy());
+    const client = new ServiceClient({
+      httpClient: {
+        sendRequest: (req) => {
+          request = req;
+          return Promise.resolve({
+            request,
+            status: 500,
+            headers: createHttpHeaders(),
+          });
+        },
+      },
+      pipeline,
+    });
+
+    let rawResponse: FullOperationResponse | undefined;
+    let requestFailed = false;
+
+    try {
+      await client.sendOperationRequest(
+        { options: { onResponse: (response) => (rawResponse = response) } },
```

Comment:
nit: should we also verify that `error` is passed to the callback?

Reply:
done, also verified flatResponse